query-connector/src/app/database-service.ts

```diff
@@ -885,3 +886,42 @@ export async function checkDBForData() {
     result.rows.length > 0 && parseFloat(result.rows[0].estimated_count) > 0
   );
 }
+
+//Cache for FHIR server configurations
+let cachedFhirServerConfigs: Promise<FhirServerConfig[]> | null = null;
+
+/**
+ * Fetches all FHIR server configurations from the database and caches the result.
+ * @returns An array of FHIR server configurations.
+ */
+export async function getFhirServerConfigs() {
+  if (cachedFhirServerConfigs) {
```

Comment:
suggestion to be taken with some salt: would we ever want an escape hatch (via an optional param?) to flush the cache?

Reply:
Since this is a server action, I think we won't be able to flush the cache just by refreshing the page, so I think Bob is right that we'll need some manual way to flush the cache

Reply:
Added a `forceRefresh` param that we can set true to manually reset. I'm thinking it might be nice to refresh when we [add/edit/delete fhir servers](https://linear.app/skylight-cdc/issue/QUE-100/add-edit-and-delete-buttons-to-front-end-fhir-server-table).